napari/components/layerlist.py

```diff
@@ -138,38 +169,18 @@ def _get_extent_world(self, layer_extent_list):
             max_v = [np.nan] * self.ndim
         else:
             extrema = [extent.world for extent in layer_extent_list]
-            mins = [e[0][::-1] for e in extrema]
-            maxs = [e[1][::-1] for e in extrema]
-
-            with warnings.catch_warnings():
-                # Taking the nanmin and nanmax of an axis of all nan
-                # raises a warning and returns nan for that axis
-                # as we have do an explict nan_to_num below this
-                # behaviour is acceptable and we can filter the
-                # warning
-                warnings.filterwarnings(
-                    'ignore',
-                    message=str(
-                        trans._('All-NaN axis encountered', deferred=True)
-                    ),
-                )
-                min_v = np.nanmin(
-                    list(itertools.zip_longest(*mins, fillvalue=np.nan)),
-                    axis=1,
-                )
-                max_v = np.nanmax(
-                    list(itertools.zip_longest(*maxs, fillvalue=np.nan)),
-                    axis=1,
-                )
+            mins = [e[0] for e in extrema]
+            maxs = [e[1] for e in extrema]
+            min_v, max_v = self._get_min_and_max(mins, maxs)
 
         try:
-            min_vals = np.nan_to_num(min_v[::-1], nan=-0.5)
-            max_vals = np.nan_to_num(max_v[::-1], nan=511.5)
+            min_vals = np.nan_to_num(min_v, nan=-0.5)
+            max_vals = np.nan_to_num(max_v, nan=511.5)
         except TypeError:
             # In NumPy < 1.17, nan_to_num doesn't have a nan kwarg
-            min_vals = np.asarray(min_v[::-1])
+            min_vals = np.asarray(min_v)
```

Comment:
We can move to 1.18+, per [NEP-29](https://numpy.org/neps/nep-0029-deprecation_policy.html)